ground/src/main/java/com/google/android/ground/ui/home/mapcontainer/PolygonDrawingViewModel.kt

```diff
@@ -154,43 +156,33 @@ class PolygonDrawingViewModel @Inject internal constructor(resources: Resources)
       CANCELED
     }
 
-    /** Current state of polygon drawing. */
-    abstract val state: State
-
-    /** Final polygon. */
-    abstract val polygon: Polygon?
-
     companion object {
-      fun canceled(): PolygonDrawingState {
-        return createDrawingState(State.CANCELED, null)
-      }
+      fun canceled(): PolygonDrawingState = PolygonDrawingState(State.CANCELED)
 
-      fun inProgress(): PolygonDrawingState {
-        return createDrawingState(State.IN_PROGRESS, null)
-      }
+      fun inProgress(): PolygonDrawingState = PolygonDrawingState(State.IN_PROGRESS)
 
-      fun completed(polygon: Polygon): PolygonDrawingState {
-        return createDrawingState(State.COMPLETED, polygon)
-      }
-
-      private fun createDrawingState(state: State, polygon: Polygon?): PolygonDrawingState {
-        return AutoValue_PolygonDrawingViewModel_PolygonDrawingState(state, polygon)
-      }
+      fun completed(polygon: Polygon): PolygonDrawingState =
+        PolygonDrawingState(State.COMPLETED, polygon)
     }
   }
 
   /** Returns a set of [Geometry] to be drawn on map for the given [Polygon]. */
-  private fun unsavedLocationsOfInterestFromLocationOfInterest(
-    polygon: Polygon
-  ): ImmutableSet<Geometry> {
+  private fun getFeaturesFromGeometries(polygon: Polygon): ImmutableSet<Feature> {
     val vertices = polygon.vertices
 
     if (vertices.isEmpty()) {
       return ImmutableSet.of()
     }
 
-    // Include the given polygon and add 1 LOI with a Point for each of its vertex.
-    return ImmutableSet.builder<Geometry>().add(polygon).addAll(vertices.toList()).build()
+    return ImmutableSet.builder<Feature>()
+      .add(
+        Feature(
+          id = uuidGenerator.generateUuid(),
+          tag = Feature.Type.LOCATION_OF_INTEREST,
```

Comment:
Should we use a different tag type for these since they aren't really LOIs? (We are writing these to submission data right? They wouldn't end up in the LOI table, which are only locations specified by the survey owners)

Reply:
From what I understand, the renderer would remain the same for LOI or data collection feature. What are we planning to use it for?

Reply:
@scolsen I think what you're calling `tag` here is really the namespace for the `id`, and the id and namespace together are what Maps SDK call a "tag". Please cmiiw.

@shobhitagarwal1612 It's used to identify the target of click and other events. Maps SDK would returns this as a Tag object, which the application layer can then use to understand if the feature refers to an LOI, user data, KML overlay, or something else.

Reply:
@scolsen Perhaps id and namespace should be grouped together as a separate FeatureTag class, since they can be returned without the geometry?

Also, if we have `Feature.Type.LOCATION_OF_INTEREST` in the view layer we're again breaking the abstraction that the map doesn't know anything about the actual different types of geometries there can be. I'd suggest defining the namespace as a `String`, and moving the enum to the view layer.

Reply:
yes, @shobhitagarwal1612, as @gino-m is suggesting the `tag` is what tells us what DB table to look at. in the map we only have opaque geometries and ID values. we could interpret these as any model object in theory, so we need to carry some information that tells us what kind of ID we're looking at/representing on the map (does this geometry represent an LOI or some other kind of entity, like user submission data that we decided we should render?)

@gino-m sgtm

Reply:
As discussed, will take the refactoring in a followup PR.